internal/tests/cluster/worker_bytesupdown_test.go

```diff
@@ -0,0 +1,131 @@
+package cluster
+
+import (
+	"context"
+	"net"
+	"testing"
+
+	"github.com/hashicorp/boundary/api/sessions"
+	"github.com/hashicorp/boundary/api/targets"
+	"github.com/hashicorp/boundary/internal/cmd/config"
+	"github.com/hashicorp/boundary/internal/daemon/controller"
+	"github.com/hashicorp/boundary/internal/daemon/worker"
+	"github.com/hashicorp/boundary/internal/tests/helper"
+	"github.com/hashicorp/dawdle"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWorkerBytesUpDown(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	logger := hclog.New(&hclog.LoggerOptions{
+		Name:  t.Name(),
+		Level: hclog.Trace,
+	})
+
+	conf, err := config.DevController()
+	require.NoError(err)
+
+	pl, err := net.Listen("tcp", "localhost:0")
+	require.NoError(err)
+	c1 := controller.NewTestController(t, &controller.TestControllerOpts{
+		Config:                    conf,
+		InitialResourcesSuffix:    "1234567890",
+		Logger:                    logger.Named("c1"),
+		PublicClusterAddr:         pl.Addr().String(),
+		StatusGracePeriodDuration: helper.DefaultGracePeriod,
+	})
+	defer c1.Shutdown()
+
+	expectWorkers(t, c1)
+
+	// Wire up the testing proxies
+	require.Len(c1.ClusterAddrs(), 1)
+	proxy, err := dawdle.NewProxy("tcp", "", c1.ClusterAddrs()[0],
+		dawdle.WithListener(pl),
+		dawdle.WithRbufSize(256),
+		dawdle.WithWbufSize(256),
+	)
+	require.NoError(err)
+	defer proxy.Close()
+	require.NotEmpty(t, proxy.ListenerAddr())
```

Comment:
Any way we can count the actual bytes going up and down this and compare it against the reported bytes?

Reply:
I don't see any methods within `dawdle.Proxy` to allow us to do this - We'd likely have to wrap the `net.Listener` with something to count `Read` and `Write` calls like we did for this task